internal/karma/karma.conf.js

```diff
@@ -3,7 +3,43 @@
 try
 {
   const fs = require('fs');
+  const path = require('path');
   const tmp = require('tmp');
+  const child_process = require('child_process');
+
+  // Helper function to find a particular namedFile
+  // within the webTestMetadata webTestFiles
+  function findNamedFile(webTestMetadata, key) {
+    let result;
+    webTestMetadata['webTestFiles'].forEach(entry => {
+      const webTestNamedFiles = entry['namedFiles'];
+      if (webTestNamedFiles && webTestNamedFiles[key]) {
+        result = webTestNamedFiles[key];
+      }
+    });
+    return result;
+  }
+
+  // Helper function to extract a browser archive
+  // and return the path to extract executable
+  function extractWebArchive(extractExe, archiveFile, executablePath) {
+    try {
+      // Paths are relative to the 'external' folder within runfiles
+      extractExe = extractExe ? path.join('external', extractExe) : extractExe;
+      archiveFile = path.join('external', archiveFile)
```

Comment:
is this going to work with the --nolegacy_external_runfiles flag?
I'll wait for that one to land before doing this one.
also missing a trailing semicolon :0

Reply:
thanks. I'll rebase to and let CI run on it,.